pkg/client/optimizely_user_context.go

```diff
@@ -26,18 +26,23 @@ import (
 
 // OptimizelyUserContext defines user contexts that the SDK will use to make decisions for.
 type OptimizelyUserContext struct {
-	optimizely  *OptimizelyClient
-	userContext entities.UserContext
-	mutex       *sync.RWMutex
+	entities.UserContext
```

Comment:
Can we keep "ID" and "attributes" here instead of UserContext? In this way, decide-api users won't see the confusion about 2 user-contexts. We can also remove the dependency onto the legacy userContext and in the long term, will have a better chance to clean up the 2 context confusion.